src/server/connection.rs

```diff
@@ -1491,8 +1495,50 @@ impl Connection {
         self.video_ack_required = lr.video_ack_required;
     }
 
+    #[cfg(target_os = "windows")]
+    async fn handle_multiple_user_sessions(&mut self, usid: Option<u32>) -> bool {
+        if self.port_forward_socket.is_some() {
+            return true;
+        } else {
+            let active_sessions = crate::platform::get_all_active_sessions();
+            if active_sessions.len() <= 1 {
+                return true;
+            }
+            let current_process_usid = crate::platform::get_current_process_session_id();
+            if usid.is_none() {
+                let mut res = Misc::new();
+                let mut rdp = Vec::new();
+                for session in active_sessions {
+                    let u_sid = &session[0];
+                    let u_name = &session[1];
+                    let mut rdp_session = RdpUserSession::new();
+                    rdp_session.user_session_id = u_sid.clone();
+                    rdp_session.user_name = u_name.clone();
+                    rdp.push(rdp_session);
+                }
+                res.set_rdp_user_sessions(RdpUserSessions {
+                    rdp_user_sessions: rdp,
+                    ..Default::default()
+                });
+                let mut msg_out = Message::new();
+                msg_out.set_misc(res);
+                self.send(msg_out).await;
+                return true;
+            }
+            if usid != Some(current_process_usid) {
+                self.on_close("Reconnecting...", false).await;
+                std::thread::spawn(move || {
+                    let _ = ipc::connect_to_user_session(usid);
+                });
+                return false;
+            }
+            true
+        }
+    }
+
     #[cfg(not(any(target_os = "android", target_os = "ios")))]
     fn try_start_cm_ipc(&mut self) {
+        let usid = self.user_session_id;
```

Comment:
If there is a console session and an rdp session, 1.2.4 choose rdp session and then close the connection, the rustdesk server is running on the rdp session, then an older version client connects, it will connect to the rdp session, but cm is on the console session. Just make cm/tray's session ID always the same as --server's session ID. function `crate::platform::get_current_process_session_id()` can be used.

Reply:
![1707094949596](https://github.com/rustdesk/rustdesk/assets/14891774/457502e1-6093-4dbc-b653-63ed117c39bd)

Switching between console and rdp for servel times, cm show in the wrong session.